snapcraft/application.py

```diff
@@ -22,87 +22,129 @@
 import pathlib
 import signal
 import sys
-from typing import Any, List, cast
+from typing import Any, List
 
 import craft_application.commands as craft_app_commands
 import craft_application.models
 import craft_cli
 import pydantic
 from craft_application import Application, AppMetadata, util
+from craft_application.errors import CraftValidationError
 from craft_application.models import BuildInfo
 from craft_cli import emit
 from craft_providers import bases
 from overrides import override
 
 from snapcraft import cli, commands, errors, models, services
 from snapcraft.commands import unimplemented
+from snapcraft.const import SUPPORTED_ARCHS
 from snapcraft.extensions import apply_extensions
-from snapcraft.models import Architecture
-from snapcraft.models.project import apply_root_packages, validate_architectures
+from snapcraft.models import Platform
+from snapcraft.models.project import apply_root_packages
 from snapcraft.providers import SNAPCRAFT_BASE_TO_PROVIDER_BASE
 from snapcraft.utils import get_effective_base, get_host_architecture
 
 
 class SnapcraftBuildPlanner(craft_application.models.BuildPlanner):
     """A project model that creates build plans."""
 
-    architectures: List[str | Architecture] = pydantic.Field(
-        default_factory=lambda: [get_host_architecture()]
-    )
-    base: str | None = None
+    base: str | None
     build_base: str | None = None
-    name: str | None = None
+    name: str
+    platforms: dict[str, Any]
```

Comment:
could this be a `dict[str, Platform]`?

Reply:
Probably a `dict[str, Platform | None]` so you can have:

```
platforms:
  amd64:
```

Reply:
I took this from [Rockcraft](https://github.com/canonical/rockcraft/blob/ac9cfd507237d64f3b619a42ec43d753c2de1ce7/rockcraft/models/project.py#L139).  If I make this too specific, I end up having to do a handlful of `if ...` or `if isinstance()` checks when making the build plan.  I'll see if I can tidy this up any more.